packages/jaeger-ui/src/model/link-patterns.js

```diff
@@ -0,0 +1,183 @@
+  }
+  return {
+    parameters,
+    template: (...args) => {
+      let text = '';
+      for (let i = 0; i < templateSplitLength; i++) {
+        if (i % 2 === 0) {
+          text += templateSplit[i];
+        } else {
+          text += encodeFn(args[templateSplit[i]]);
+        }
+      }
+      return text;
+    },
+  };
+}
+
+export function createTestFunction(entry) {
+  if (typeof entry === 'string') {
+    return arg => arg === entry;
+  }
+  if (Array.isArray(entry)) {
+    return arg => entry.indexOf(arg) > -1;
+  }
+  if (entry instanceof RegExp) {
+    return arg => entry.test(arg);
+  }
+  if (typeof entry === 'function') {
+    return entry;
+  }
+  if (entry == null) {
+    return () => true;
+  }
+  throw new Error(`Invalid value: ${entry}`);
+}
+
+const identity = a => a;
+
+export function processLinkPattern(pattern) {
+  try {
+    const url = processTemplate(pattern.url, encodeURIComponent);
+    const text = processTemplate(pattern.text, identity);
+    return {
+      object: pattern,
+      type: createTestFunction(pattern.type),
+      key: createTestFunction(pattern.key),
+      value: createTestFunction(pattern.value),
+      url,
+      text,
+      parameters: _uniq(url.parameters.concat(text.parameters)),
+    };
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(`Ignoring invalid link pattern: ${error}`, pattern);
+    return null;
+  }
+}
+
+export function getParameterInArray(name, array) {
+  if (array) {
+    return array.find(entry => entry.key === name);
+  }
+  return undefined;
+}
+
+export function getParameterInAncestor(name, spans, startSpanIndex) {
+  let currentSpan = { depth: spans[startSpanIndex].depth + 1 };
+  for (let spanIndex = startSpanIndex; spanIndex >= 0; spanIndex--) {
+    const nextSpan = spans[spanIndex];
+    if (nextSpan.depth < currentSpan.depth) {
+      currentSpan = nextSpan;
+      const result =
+        getParameterInArray(name, currentSpan.tags) || getParameterInArray(name, currentSpan.process.tags);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return undefined;
+}
+
+export function callTemplate(template, data) {
+  return template.template(...template.parameters.map(param => data[param]));
+}
+
+export function computeLinks(linkPatterns, trace, spanIndex, items, itemIndex) {
+  const item = items[itemIndex];
+  const span = trace.spans[spanIndex];
+  let type = 'logs';
+  const processTags = span.process.tags === items;
+  if (processTags) {
+    type = 'process';
+  }
+  const spanTags = span.tags === items;
+  if (spanTags) {
+    type = 'tags';
+  }
+  const result = [];
+  linkPatterns.forEach(pattern => {
+    if (pattern.type(type) && pattern.key(item.key, item.value, type) && pattern.value(item.value)) {
+      let parameterValues = {};
+      pattern.parameters.every(parameter => {
+        let entry = getParameterInArray(parameter, items);
+        if (!entry && !processTags) {
```

Comment:
I believe ancestors are also being searched when type === 'log', which seems to contradict your comment on the PR description?